backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -226,3 +277,127 @@ def revoke_share(cls, engine: Engine, share_uri: str):
             share_sm.update_state(session, share, new_share_state)
 
             return revoked_folders_succeed and revoked_s3_buckets_succeed and revoked_tables_succeed
+
+    @staticmethod
+    def acquire_lock(dataset_uri, session, share_uri):
+        """
+        Attempts to acquire a lock on the dataset identified by dataset_id.
+
+        Args:
+            dataset_uri: The ID of the dataset for which the lock is being acquired.
+            session (sqlalchemy.orm.Session): The SQLAlchemy session object used for interacting with the database.
+            share_uri: The ID of the share that is attempting to acquire the lock.
+
+        Returns:
+            bool: True if the lock is successfully acquired, False otherwise.
+        """
+        try:
+            # Execute the query to get the DatasetLock object
+            dataset_lock = (
+                session.query(DatasetLock)
```

Comment:
Can there be a situation in which two ECS processes start at the same time and arrive here and read the same state . And then update the dataset_lock table on L308 - 311. 

Similar to how it is mentioned here - https://dev.to/ivankwongtszfung/safe-update-operation-in-postgresql-using-sqlalchemy-3ela

Reply:
Reading the above docs it seems like this is true - should we include `.with_for_update().first()` as specified in the docs @TejasRGitHub posted cc: @anushka-singh 

I believe that will resolve as no other session can update that record and will return an OperationalError that we can handle in exceptions

Reply:
@TejasRGitHub, In my code, I had added .first() to fetch only the first entry. So, I dont think a process will be able to read 2 states at the same time. 

I have anyway added the `.with_for_update().first()` guardrail to be safe. Thanks for the suggestion!